README.md

```diff
@@ -8,43 +8,105 @@ This repo builds a Docker image which collects tooling used to setup environment
-IMAGE="ghcr.io/xenitab/github-actions/tools:latest"
-OPA_BLAST_RADIUS := $(if $(OPA_BLAST_RADIUS), $(OPA_BLAST_RADIUS), 50)
-AZURE_CONFIG_DIR := $(if $(AZURE_CONFIG_DIR), $(AZURE_CONFIG_DIR), "$${HOME}/.azure")
+SUFFIX="tfstate[random 4 digits]"
+IMAGE="ghcr.io/xenitab/github-actions/tools:[tag]"
+AWS_ENABLED:=false
+
+OPA_BLAST_RADIUS := $(if $(OPA_BLAST_RADIUS),$(OPA_BLAST_RADIUS),50)
+AZURE_CONFIG_DIR := $(if $(AZURE_CONFIG_DIR),$(AZURE_CONFIG_DIR),"$${HOME}/.azure")
 TTY_OPTIONS=$(shell [ -t 0 ] && echo '-it')
+TEMP_ENV_FILE:=$(shell mktemp)
+
 ifndef ENV
 $(error Need to set ENV)
 endif
 ifndef DIR
 $(error Need to set DIR)
 endif
 
-prepare:
-        docker run --user $(shell id -u) $(TTY_OPTIONS) --entrypoint "/opt/terraform.sh" -v $${PWD}/$(DIR)/.terraform/$(ENV).env:/tmp/$(ENV).env -v $(AZURE_CONFIG_DIR):/work/.azure -v $${PWD}/$(DIR):/tmp/$(DIR) -v $${PWD}/global.tfvars:/tmp/global.tfvars $(IMAGE) prepare $(DIR) $(ENV) $(SUFFIX)
-
-plan:
-        docker run --user $(shell id -u) $(TTY_OPTIONS) --entrypoint "/opt/terraform.sh" -v $${PWD}/$(DIR)/.terraform/$(ENV).env:/tmp/$(ENV).env -v $(AZURE_CONFIG_DIR):/work/.azure -v $${PWD}/$(DIR):/tmp/$(DIR) -v $${PWD}/global.tfvars:/tmp/global.tfvars $(IMAGE) plan $(DIR) $(ENV) $(SUFFIX) $(OPA_BLAST_RADIUS)
-
-apply:
-        docker run --user $(shell id -u) $(TTY_OPTIONS) --entrypoint "/opt/terraform.sh" -v $${PWD}/$(DIR)/.terraform/$(ENV).env:/tmp/$(ENV).env -v $(AZURE_CONFIG_DIR):/work/.azure -v $${PWD}/$(DIR):/tmp/$(DIR) -v $${PWD}/global.tfvars:/tmp/global.tfvars $(IMAGE) apply $(DIR) $(ENV) $(SUFFIX)
-
-destroy:
-        docker run --user $(shell id -u) $(TTY_OPTIONS) --entrypoint "/opt/terraform.sh" -v $${PWD}/$(DIR)/.terraform/$(ENV).env:/tmp/$(ENV).env -v $(AZURE_CONFIG_DIR):/work/.azure -v $${PWD}/$(DIR):/tmp/$(DIR) -v $${PWD}/global.tfvars:/tmp/global.tfvars $(IMAGE) destroy $(DIR) $(ENV) $(SUFFIX)
-
-state-remove:
-        docker run --user $(shell id -u) $(TTY_OPTIONS) --entrypoint "/opt/terraform.sh" -v $${PWD}/$(DIR)/.terraform/$(ENV).env:/tmp/$(ENV).env -v $(AZURE_CONFIG_DIR):/work/.azure -v $${PWD}/$(DIR):/tmp/$(DIR) -v $${PWD}/global.tfvars:/tmp/global.tfvars $(IMAGE) state-remove $(DIR) $(ENV) $(SUFFIX)
-
-validate:
-        docker run --user $(shell id -u) $(TTY_OPTIONS) --entrypoint "/opt/terraform.sh" -v $${PWD}/$(DIR)/.terraform/$(ENV).env:/tmp/$(ENV).env -v $(AZURE_CONFIG_DIR):/work/.azure -v $${PWD}/$(DIR):/tmp/$(DIR) -v $${PWD}/global.tfvars:/tmp/global.tfvars $(IMAGE) validate $(DIR) $(ENV) $(SUFFIX)
+AZURE_DIR_MOUNT:=$(shell echo "-v $(AZURE_CONFIG_DIR):/work/.azure")
+COMMAND:=$(shell echo "docker run --user $(shell id -u) $(TTY_OPTIONS) --entrypoint "/opt/terraform.sh" --env-file $(TEMP_ENV_FILE) $(AZURE_DIR_MOUNT) -v $${PWD}/$(DIR):/tmp/$(DIR) -v $${PWD}/global.tfvars:/tmp/global.tfvars $(IMAGE)")
+CLEANUP_COMMAND:=$(shell echo "$(MAKE) --no-print-directory teardown TEMP_ENV_FILE=$(TEMP_ENV_FILE)")
+
+.PHONY: setup
+.SILENT: setup
+setup:
```

Comment:
Couldn't this target be included in the container image? E.g. 
```
docker run --entrypoint /opt/printauth-tool.sh $(IMAGE) | source
printauth > $(TEMP_ENV_FILE)
```
or even just
```
docker run --entrypoint /bin/bash $(IMAGE) -c "cat /opt/printauth.sh" | source
printauth > $(TEMP_ENV_FILE)
```
?
It seems like it would be identical across usages and could benefit from standardization.

Reply:
Love the idea, but let's put that on the back burner until the new format has stabilized.